roles/ceph-osd/tasks/scenarios/lvm.yml

```diff
@@ -1,164 +1,17 @@
-  command: "ceph-volume --cluster {{ cluster }} lvm create --filestore --data {{ item.data }} --journal {{item.journal_vg }}/{{ item.journal }}"
-  environment:
-    CEPH_VOLUME_DEBUG: 1
-  with_items: "{{ lvm_volumes }}"
-  when:
-    - item.journal_vg is defined
-    - item.data_vg is not defined
-    - osd_objectstore == 'filestore'
-    - "'{{ item.data }}' not in ceph_volume_lvm_list.stdout"
-
-- name: use ceph-volume to create bluestore osds with db and wal and a raw device or partition for data
-  command: "ceph-volume --cluster {{ cluster }} lvm create --bluestore --data {{ item.data }} --block.wal {{ item.wal_vg }}/{{ item.wal }} --block.db {{ item.db_vg }}/{{ item.db }}"
-  environment:
-    CEPH_VOLUME_DEBUG: 1
-  with_items: "{{ lvm_volumes }}"
-  when:
-    - osd_objectstore == 'bluestore'
-    - item.data_vg is not defined
-    - item.wal is defined
-    - item.wal_vg is defined
-    - item.db is defined
-    - item.db_vg is defined
-    - "'{{ item.data }}' not in ceph_volume_lvm_list.stdout"
-
-- name: use ceph-volume to create bluestore osds with db only and a raw device or partition for data
-  command: "ceph-volume --cluster {{ cluster }} lvm create --bluestore --data {{ item.data }} --block.db {{ item.db_vg }}/{{ item.db }}"
-  environment:
-    CEPH_VOLUME_DEBUG: 1
-  with_items: "{{ lvm_volumes }}"
-  when:
-    - osd_objectstore == 'bluestore'
-    - item.data_vg is not defined
-    - item.wal is not defined
-    - item.wal_vg is not defined
-    - item.db is defined
-    - item.db_vg is defined
-    - "'{{ item.data }}' not in ceph_volume_lvm_list.stdout"
-
-- name: use ceph-volume to create bluestore osds with wal only and a raw device or partition for data
-  command: "ceph-volume --cluster {{ cluster }} lvm create --bluestore --data {{ item.data }} --block.wal {{ item.wal_vg }}/{{ item.wal }}"
-  environment:
-    CEPH_VOLUME_DEBUG: 1
-  with_items: "{{ lvm_volumes }}"
-  when:
-    - osd_objectstore == 'bluestore'
-    - item.wal is defined
-    - item.data_vg is not defined
-    - item.wal_vg is defined
-    - item.db is not defined
-    - item.db_vg is not defined
-    - "'{{ item.data }}' not in ceph_volume_lvm_list.stdout"
-
-- name: use ceph-volume to create bluestore osds with just a data device and a raw device or partition for data
-  command: "ceph-volume --cluster {{ cluster }} lvm create --bluestore --data {{ item.data }}"
+- name: use ceph-volume to create osds
+  ceph_volume:
+    cluster: "{{ cluster }}"
+    objectstore: "{{ osd_objectstore }}"
+    data: "{{ item.data }}"
+    data_vg: "{{ item.data_vg|default(omit) }}"
+    journal: "{{ item.journal|default(omit) }}"
+    journal_vg: "{{ item.journal_vg|default(omit) }}"
+    db: "{{ item.db|default(omit) }}"
+    db_vg: "{{ item.db_vg|default(omit) }}"
+    wal: "{{ item.wal|default(omit) }}"
+    wal_vg: "{{ item.wal_vg|default(omit) }}"
   environment:
     CEPH_VOLUME_DEBUG: 1
   with_items: "{{ lvm_volumes }}"
```

Comment:
shouldn't this be `devices` at some point since we support raw block devices on ceph-volume?

Reply:
No, we use ``lvm_volumes`` for anything ``ceph-volume``. A raw block device is accepted in a ``data`` param in ``lvm_volumes`` though.